### PR TITLE
Configurable numeric precision for Diagnostics panel values

### PR DESCRIPTION
**Pull request by jhurliman**
**User-Facing Changes**
- Diagnostics Details panel now supports configurable numeric precision

**Description**
Screen Shot 2022-10-26 at 4 08 40 PM

### CODE REVIEW

File: packages/studio-base/src/panels/diagnostics/DiagnosticStatus.tsx [hunk 1]
```diff
@@ -42,6 +42,7 @@ type Props = {
   splitFraction: number | undefined;
   onChangeSplitFraction: (arg0: number) => void;
   topicToRender: string;
+  precision: number | undefined;
```
Comment (defunctzombie) [inline packages/studio-base/src/panels/diagnostics/DiagnosticStatus.tsx:45]: Let's change this variable to `numericPrecision` to be more specific and match the configuration label.

**Review by defunctzombie — COMMENTED** [review]
The default, auto, and 0 behavior feels awkward.

Here's what I see when I opened the preview deployment for the first time:

image

I think this means "auto"? But I don't know.

After changing the precision and then going back to `0` I see the following:

image

I have no way to go back to "auto" now.

File: packages/studio-base/src/panels/diagnostics/DiagnosticStatus.tsx [hunk 2]
```diff
@@ -219,6 +221,15 @@ export default function DiagnosticStatus(props: Props): JSX.Element {
       if (html) {
         return <HTMLTableCell dangerouslySetInnerHTML={html} />;
       }
+
+      let strToRender = str;
+      if (precision != undefined) {
+        const maybeFloat = parseFloat(strToRender);
```
Comment (jtbandes) [inline packages/studio-base/src/panels/diagnostics/DiagnosticStatus.tsx:227]: So if someone has an entry with the string “0x12”, and they set numeric precision, that’ll turn into “0.00”?

**Comment by jhurliman** [comment]
> I have no way to go back to "auto" now.

@defunctzombie you go back to "auto" by deleting the input value. This allows you to differentiate between "auto" and `0` which have different behavior (full precision vs integer rounding).

**Comment by defunctzombie** [comment]
> @defunctzombie you go back to "auto" by deleting the input value. This allows you to differentiate between "auto" and `0` which have different behavior (full precision vs integer rounding).

Makes sense. Would be good for the display to show "auto" when that happens or when there is no value. I believe we do this in other panels.

**Comment by jhurliman** [comment]
@defunctzombie comments addressed, can you take another look?

File: packages/studio-base/src/panels/diagnostics/DiagnosticStatus.tsx [hunk 3]
```diff
@@ -368,3 +378,14 @@ export default function DiagnosticStatus(props: Props): JSX.Element {
...
+// Returns true if the input string can be parsed as a float or an integer using
+// parseFloat(). Hex and octal numbers will return false.
+function isFloatOrInteger(n: string): boolean {
+  if (n[0] === "0" && n.length > 1) {
+    if (n[1] === "x" || n[1] === "X" || n[1] === "o" || n[1] === "O") {
+      return false;
+    }
+  }
+  return !isNaN(parseFloat(n)) && isFinite(Number(n));
+}
```
Comment (jtbandes) [inline packages/studio-base/src/panels/diagnostics/DiagnosticStatus.tsx:391]: 👍 Unit tests please

**Review by defunctzombie — APPROVED** [review]
Feels much better! Please add the requested unit test and LGTM.

**Pull request merged**